test/integration/authz_v1_test.go

```diff
@@ -0,0 +1,388 @@
+		assert.Nil(t, err, "error should be nil")
+		r.AD.ResourceVersion = currentAD.ResourceVersion
+		_, err = framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Update(r.AD)
+		assert.Nil(t, err, "error should be nil")
+	} else if a == create {
+		_, err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Create(r.AD)
+		assert.Nil(t, err, "should be nil")
+	}
+
+	err := wait.PollImmediate(time.Second, time.Second*5, func() (bool, error) {
+		for _, curr := range r.ModelConfigs {
+			got := framework.Global.IstioClientset.Get(curr.Type, curr.Name, curr.Namespace)
+			if got == nil {
+				return false, nil
+			}
+
+			if !reflect.DeepEqual(curr.Spec, got.Spec) {
+				return false, nil
+			}
+		}
+
+		return true, nil
+	})
+
+	if err != nil {
+		t.Error("time out waiting for rollout for ad", r.AD.Name, "with error", err)
+	}
+
+	validateConfigs(t, r)
+}
+
+// validateConfigs will validate the service role / service role bindings on the
+// cluster against the expected output
+func validateConfigs(t *testing.T, r *fixtures.ExpectedResources) {
+	for _, curr := range r.ModelConfigs {
+		got := framework.Global.IstioClientset.Get(curr.Type, curr.Name, curr.Namespace)
+		assert.NotNil(t, got, "istio custom resource should exist on the cluster")
+		curr.ResourceVersion = got.ResourceVersion
+		curr.CreationTimestamp = got.CreationTimestamp
+		assert.Equal(t, curr, *got, "istio resource should be equal to expected")
+	}
+}
+
+// cleanup will clean up the athenz domain and service role / service role binding objects on the cluster
+func cleanup(t *testing.T, r *fixtures.ExpectedResources) {
+	err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Delete(r.AD.Name, &v1.DeleteOptions{})
+	assert.Nil(t, err, "athenz domain delete error should be nil")
+
+	for _, curr := range r.ModelConfigs {
+		err := framework.Global.IstioClientset.Delete(curr.Type, curr.Name, curr.Namespace)
+		assert.Nil(t, err, "istio custom resource delete error should be nil")
+	}
+}
+
+// 1.0 Create SR / SRB with valid AD
+func TestCreateServiceRoleAndBinding(t *testing.T) {
+	r := fixtures.GetExpectedResources(&fixtures.OverrideResources{})
+	rolloutAndValidate(t, r, create)
+	cleanup(t, r)
+}
+
+// 1.1 Create SR only if there are no role members for SRB creation
+func TestCreateServiceRoleOnly(t *testing.T) {
+	o := &fixtures.OverrideResources{
+		ModifyAD: func(signedDomain *zms.SignedDomain) {
+			signedDomain.Domain.Roles[0].RoleMembers = []*zms.RoleMember{}
+			signedDomain.Domain.Roles[0].Members = []zms.MemberName{}
+		},
+	}
+	r := fixtures.GetExpectedResources(o)
+	rolloutAndValidate(t, r, create)
+	srb := framework.Global.IstioClientset.Get(model.ServiceRoleBinding.Type, "client-writer-role", r.AD.Namespace)
+	assert.Nil(t, srb, "service role binding should not exist")
+	cleanup(t, r)
+}
+
+// 2.0 Update existing role / policy
```

Comment:
should this be `Update existing domain with new role / policy`?

Reply:
Fixed.